WebDriverAgentLib/FBApplication.m

```diff
@@ -122,7 +122,7 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
     return;
   }
   XCUIApplicationProcess *applicationProcess = change[NSKeyValueChangeNewKey];
-  if (!applicationProcess || ![applicationProcess isMemberOfClass:XCUIApplicationProcess.class]) {
+  if (!applicationProcess || [applicationProcess isProxy] || ![applicationProcess isMemberOfClass:XCUIApplicationProcess.class]) {
```

Comment:
I don't think that explicitly checking for proxy here is what we want, as we shouldn't be even thinking about it being a proxy. `isMemberOfClass` should be also forwarded to `XCUIApplicationProcess` and would return YES , which is what we want right? So the `if` statement would behave correctly here.

Reply:
If remove check for ```[applicationProcess isProxy]``` then execution of the method ```observeValueForKeyPath``` would continue, and the line of code
```
[object setValue:[FBApplicationProcessProxy proxyWithApplicationProcess:applicationProcess] forKey:keyPath];
```
gets executed. 

if the ```applicationProcess``` is a proxy object (```FBApplicationProcessProxy```), then it would instantiate a new proxy object (```FBApplicationProcessProxy```) with parameter ```applicationProcess``` (which is a proxy itself already).

This happens in rare cases , when in one test several applications are used (several WebDriver sessions) — for example: 1. Start session with Safari, 2. Stop session, 3. Start session with Preferences, 4. Stop session, 5. Start session with Safari, 2. Stop session

In this case it would start a recursive calls to proxied objects.

Reply:
You are absolutely right, I completely missed the point of that `if` ;P.